src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -567,23 +590,25 @@ export const VaultDepositWithdrawForm = ({
       <$FlexFill />
 
       <WithDetailsReceipt detailItems={inputFormConfig.receiptItems}>
-        <Button
-          type={ButtonType.Submit}
-          action={ButtonAction.Primary}
-          state={{
-            isDisabled: hasInputErrors || !!isAccountViewOnly || !canViewAccount || isSubmitting,
-            isLoading: isSubmitting,
-          }}
-          slotLeft={
-            errorsPreventingSubmit.find((f) => !lightErrorKeys.has(f.code)) != null ? (
-              <$WarningIcon iconName={IconName.Warning} />
-            ) : undefined
-          }
-        >
-          {hasInputErrors && errorsPreventingSubmit[0]?.short != null
-            ? errorsPreventingSubmit[0]?.short
-            : inputFormConfig.buttonLabel}
-        </Button>
+        {maybeConnectWalletButton ?? (
```

Comment:
would a ternary using `const ctaButton = canAccountTrade ? ... : ...` be easier to read than setting `maybeConnectWalletButton` w/ the nullish check.

Reply:
ah nvm. I see that `maybeConnectWalletButton is used in another nullish coalescing op below

Reply:
Tina said the same but my intent is to centralize the logic for "should I display this" and "what to display" in one place since it's rendered twice. If you guys still think it's better to copy+paste the ternary I'm down to but I think this is slightly better given the tradeoffs.